src/MeetingsSDKAdapter.test.js

```diff
@@ -0,0 +1,121 @@
+import * as rxjs from 'rxjs';
+
+import MeetingSDKAdapter from './MeetingsSDKAdapter';
+import createMockSDK from './__mocks__/sdk';
+
+describe('Meetings SDK Adapter', () => {
+  let meetingSDKAdapter, mockSDK, meetingID, meeting, target;
+
+  beforeEach(() => {
+    mockSDK = createMockSDK();
+    meetingSDKAdapter = new MeetingSDKAdapter(mockSDK);
+    meetingID = 'meetingID';
+    rxjs.fromEvent = jest.fn(() => rxjs.of({}));
+    meeting = {
+      ID: meetingID,
+      localAudio: null,
+      localShare: null,
+      localVideo: null,
+      remoteShare: null,
+      remoteStream: null,
+      title: 'my meeting',
+    };
+    target = 'target';
+  });
+
+  afterEach(() => {
+    meeting = null;
+    rxjs.fromEvent = null;
+    mockSDK = null;
+    meetingSDKAdapter = null;
+    meetingID = null;
+    target = null;
+  });
+
+  describe('attachMedia()', () => {
+    test('sets `localAudio` and `localVideo`, if the event type is `local`', () => {
+      const event = {
+        type: 'local',
+        stream: {
+          getAudioTracks: jest.fn(() => ['localAudio']),
+          getVideoTracks: jest.fn(() => ['localVideo']),
+        },
+      };
+
+      meetingSDKAdapter.attachMedia(meetingID, event);
+      expect(meetingSDKAdapter.meetings[meetingID]).toMatchObject({
+        localAudio: 'localAudio',
+        localVideo: 'localVideo',
+      });
+    });
+
+    test('returns the same meeting object, if the event type is not declared', () => {
+      meetingSDKAdapter.meetings[meetingID] = meeting;
+      const event = {
+        type: 'NA',
+        stream: {},
+      };
+
+      meetingSDKAdapter.attachMedia(meetingID, event);
+      expect(meetingSDKAdapter.meetings[meetingID]).toMatchObject(meeting);
+    });
+  });
+
+  describe('createMeeting()', () => {
+    test('returns a new meeting in a proper shape', (done) => {
+      meetingSDKAdapter.createMeeting(target).subscribe((newMeeting) => {
+        expect(newMeeting).toMatchObject(meeting);
+        done();
+      });
+    });
+
+    test('throws error on failed meeting push request', (done) => {
+      const wrongTarget = 'wrongTarget';
+      const errorMessage = `Unable to create a meeting "${wrongTarget}"`;
+
+      meetingSDKAdapter.datasource.meetings.create = jest.fn(() => Promise.reject(errorMessage));
+      meetingSDKAdapter.createMeeting(target).subscribe(
+        () => {},
+        (error) => {
+          expect(error).toBe(errorMessage);
+          done();
+        }
+      );
+    });
+  });
+
+  describe('getMeeting()', () => {
+    test('returns a meeting in a proper shape', (done) => {
+      meetingSDKAdapter.createMeeting(target).subscribe(() => {
```

Comment:
Is this the expected usage for `getMeeting`? If so, I think we should encourage piping instead of nesting subscriptions.
```
meetingSDKAdapter.createMeeting(target).pipe(
  flatMap(({ID}) => meetingSDKAdapter.getMeeting(ID))
)
.subscribe(...);